test/integration/test/pagekite_access_test.js

```diff
@@ -0,0 +1,142 @@
+'use strict';
+
+const spawn = require('child_process').spawn;
+const chakram = require('chakram'), expect = chakram.expect;
+const Config = require('config-js');
+
+var philipshue_server = require('../lib/philipsHue_server.js');
+var nupnp_server = require('../lib/nupnp_PhilipsHue.js');
+var config = new Config('./test/integration/lib/config/foxbox.js');
+var header = new Config('./test/integration/lib/config/header.js');
+
+describe('Initiate the remote connection',function(){
+  var credential = config.get('credential'); 
+  var lightinfo; // to store the light status received from foxbox
+  var foxbox_process;
+
+  var setupURL = config.get('foxbox.url') + '/users/setup';
+  var loginURL = 'http://' + config.get('pagekite.externalURL')+ '/users/login';
+  var serviceURL = 'http://' + config.get('pagekite.externalURL')+ '/services/';
+  var serviceListURL = serviceURL + 'list';  
+  
+  before(function(done){
+
+    var FOXBOX_STARTUP_WAIT_TIME_IN_MS = 3000;
+    this.timeout(500000);
+
+    // start nupnp server
+    nupnp_server.start(config.get('nupnp_server.id'),
```

Comment:
As the nupnp_server is started the same way in the 3 tests, I'd put the default configuration in that module.